sdk/metrics/src/test/java/io/opentelemetry/sdk/metrics/internal/state/SynchronousMetricStorageTest.java

```diff
@@ -59,44 +64,56 @@ public Attributes onLabelsBound(Context ctx, Attributes lbls) {
-    AggregatorHandle<?> aggregatorHandle = accumulator.bind(labels);
-    aggregatorHandle.recordDouble(1);
-    List<MetricData> md = accumulator.collectAll(testClock.now());
-    md.stream()
-        .flatMap(m -> m.getLongGaugeData().getPoints().stream())
-        .forEach(
-            p -> assertThat(p.getAttributes()).hasSize(1).containsEntry("modifiedK", "modifiedV"));
+    SynchronousMetricStorage<?> accumulator =
+        new SynchronousMetricStorage<>(
+            METRIC_DESCRIPTOR,
+            aggregator,
+            new InstrumentProcessor<>(aggregator, testClock.now()),
+            spyLabelsProcessor);
+    BoundStorageHandle handle = accumulator.bind(labels);
+    handle.recordDouble(1, labels, Context.root());
+    MetricData md = accumulator.collectAndReset(0, testClock.now());
+    assertThat(md)
+        .hasDoubleGauge()
+        .points()
+        .allSatisfy(
+            p ->
+                assertThat(p)
+                    .attributes()
+                    .hasSize(2)
+                    .containsEntry("modifiedK", "modifiedV")
+                    .containsEntry("K", "V"));
   }
 
   @Test
-  void sameAggregator_ForSameLabelSet() {
-    SynchronousInstrumentAccumulator<?> accumulator =
-        new SynchronousInstrumentAccumulator<>(
-            aggregator, new InstrumentProcessor<>(aggregator, testClock.now()), labelsProcessor);
-    AggregatorHandle<?> aggregatorHandle =
-        accumulator.bind(Attributes.builder().put("K", "V").build());
-    AggregatorHandle<?> duplicateAggregatorHandle =
+  void sameAggregator_ForSameAttributes() {
+    SynchronousMetricStorage<?> accumulator =
+        new SynchronousMetricStorage<>(
+            METRIC_DESCRIPTOR,
+            aggregator,
+            new InstrumentProcessor<>(aggregator, testClock.now()),
+            labelsProcessor);
+    BoundStorageHandle handle = accumulator.bind(Attributes.builder().put("K", "V").build());
+    BoundStorageHandle duplicateAggregatorHandle =
         accumulator.bind(Attributes.builder().put("K", "V").build());
     try {
-      assertThat(duplicateAggregatorHandle).isSameAs(aggregatorHandle);
-      accumulator.collectAll(testClock.now());
-      AggregatorHandle<?> anotherDuplicateAggregatorHandle =
+      assertThat(duplicateAggregatorHandle).isSameAs(handle);
+      accumulator.collectAndReset(0, testClock.now());
+      BoundStorageHandle anotherDuplicateAggregatorHandle =
           accumulator.bind(Attributes.builder().put("K", "V").build());
       try {
-        assertThat(anotherDuplicateAggregatorHandle).isSameAs(aggregatorHandle);
+        assertThat(anotherDuplicateAggregatorHandle).isSameAs(handle);
       } finally {
         anotherDuplicateAggregatorHandle.release();
       }
     } finally {
       duplicateAggregatorHandle.release();
-      aggregatorHandle.release();
+      handle.release();
     }
 
+    // TODO: Is it necessary to test this?
```

Comment:
is there a problem with testing it?

Reply:
It's an implementation detail.  Is it worth exposing this interface on the Storage class to test this?  

Additionally there are already tests in place that make sure if an Attribute-set is not reported within the interval that it stops showing up.  I don't think we need to also check the contract of the `tryUnmap` method here.

Reply:
ah, I see. If removing this would let us narrow the interface, then 👍🏽